integration/tests/cook/test_basic.py

```diff
@@ -337,3 +340,42 @@ def test_application_field(self):
         # Should require application version
         _, resp = self.submit_job(application={'name': 'foo-app'})
         self.assertEqual(resp.status_code, 400)
+
+    def test_allow_partial(self):
```

Comment:
Yes, the scenarios in this test include both absence of the `partial` param and setting it explicitly to false. Let me know what you think.